website/blog/four-approaches-for-documenting-apis.md

```diff
@@ -0,0 +1,126 @@
+
+### How It Works
+
+Markdown is a highly supported text format documentation language that allows for hyperlinks and other HTML-type formatting while retaining a plain text format. It can be stored alongside code files in repositories and used with tools such as [Slate](https://docs.slatejs.org/) and [Apiary](https://apiary.io/) to create static websites the entire team can use to access API documentation. Many popular code editors such as VSCode and GitLab support Markdown files. 
+
+### Pros 
+
+**Full control:** Updating docs manually is an easy-to-use approach that gives you full control over your documentation. Documentation is presented in a simple format that is easier to understand than generated API documentation.
+
+**No tooling required:** Developers can add code comments or annotations as they work without having to switch to separate tools. 
+
+### Cons 
+
+**Time-consuming:** Manually writing documentation can be time-consuming, especially if your API is on the larger side. The prospect of creating and maintaining large volumes of API documentation can be daunting for developers. 
+
+**Annoying:** Most developers don’t enjoy creating documentation. Because developers are more eager to get on with coding, inline documentation is likely to be sparse and filled with technical jargon that might not be helpful to the average API consumer.
+
+**Hard to keep accurate:** Keeping documents in sync with code is also a major challenge using this process. Developers don’t want to spend time constantly updating API documentation. Eventually, the documentation and the code will fall out of sync and become frustrating for anyone attempting to use it. 
+
+**Prone to errors:** Developers may change API contracts, causing them to break on the consumer end, create security issues, or create a roadblock for other team members. 
+
+## Method 2: Design First
+
+Designing and writing your API definition before writing any code, and then building to that design, is referred to as the Design First approach. 
+
+### How It Works
+
+With the Design First approach, the specifics of your API—including endpoints, requests, response objects, error codes, and headers—are fleshed out in detail and tested before being handed over to developers for implementation. 
+
+After the design is complete, you can share API documents with other technical and non-technical stakeholders for review and feedback. This gives you the opportunity to catch issues before development begins. Once the design is approved, developers can use it to write or generate code implementations using tools such as [Swagger Editor](http://editor.swagger.io/# “Swagger Editor open source editor to design, define, and document RESTful APIs in the Swagger Specification”). 
```

Comment:
Tweaked this in upcoming commit - Docusaurus doesn't recognize title strings in Markdown links